tests/unit_tests/test_helm_chart.py

```diff
@@ -0,0 +1,228 @@
+    )
+    assert rendered_config == worker_config
+
+
+@pytest.mark.parametrize(
+    "values",
+    [
+        {
+            "initContainer": {
+                "scratch": {
+                    "repositories": [],
+                    "root": "/blueapi-plugins/scratch",
+                }
+            }
+        },
+        {
+            "initContainer": {
+                "scratch": {
+                    "root": "/dls_sw/i22/scratch",
+                    "required_gid": 12345,
+                    "repositories": [
+                        {
+                            "name": "foo",
+                            "remote_url": "https://example.git",
+                        },
+                        {
+                            "name": "bar",
+                            "remote_url": "https://example.git",
+                        },
+                    ],
+                }
+            }
+        },
+    ],
+)
+def test_helm_chart_creates_init_config_map(values: Values):
+    manifests = render_chart(values=values)
+    rendered_config = yaml.safe_load(
+        manifests["blueapi-initconfig"]["data"]["initconfig.yaml"]
+    )
+    assert rendered_config == values["initContainer"]
+
+
+def test_helm_chart_does_not_render_arbitrary_rabbitmq_password():
+    manifests = render_chart(
+        values={"worker": {"stomp": {"auth": {"password": "foobar"}}}}
+    )
+    rendered_config = ApplicationConfig(
+        **yaml.safe_load(manifests["blueapi-config"]["data"]["config.yaml"])
+    )
+    assert rendered_config.stomp is not None
+    assert rendered_config.stomp.auth is not None
+    assert isinstance(rendered_config.stomp.auth.password, Secret)
+
+
+def test_container_gets_container_resources():
+    manifests = render_chart(values={"resources": HIGH_RESOURCES})
+    assert (
+        manifests["blueapi"]["spec"]["template"]["spec"]["containers"][0]["resources"]
+        == HIGH_RESOURCES
+    )
+
+
+@pytest.mark.xfail(
+    reason="https://github.com/DiamondLightSource/blueapi/issues/646",
+    strict=True,
+)
+def test_init_container_gets_container_resources_by_default():
+    manifests = render_chart(
+        values={
+            "resources": HIGH_RESOURCES,
+            "scratchHostPath": "/foo",
+            "initContainer": {"scratch": {"root": "/foo"}},
+        }
+    )
+    assert (
+        manifests["blueapi"]["spec"]["template"]["spec"]["initContainers"][0][
+            "resources"
+        ]
+        == HIGH_RESOURCES
+    )
+
+
+@pytest.mark.xfail(
+    reason="https://github.com/DiamondLightSource/blueapi/issues/646",
+    strict=True,
+)
+def test_init_container_resources_overridable():
+    manifests = render_chart(
+        values={
+            "resources": HIGH_RESOURCES,
+            "initResources": LOW_RESOURCES,
+            "scratchHostPath": "/foo",
+        }
+    )
+    assert (
+        manifests["blueapi"]["spec"]["template"]["spec"]["initContainers"][0][
```

Comment:
The Service, Ingress, StatefulSet all have  `name: {{ include "blueapi.fullname" . }}`- is `service.enabled` false by default?

Reply:
Grouped rendered manifests by kind and then by name